alica_engine/src/engine/teammanager/TeamManager.cpp

```diff
@@ -97,13 +97,13 @@ void TeamManager::readSelfFromConfig(const YAML::Node& config)
     }
     const std::string localAgentName = _engine->getLocalAgentName();
 
-    if (_localAgentID == nullptr) {
+    if (_localAgentID == 0) {
         constexpr auto notAValidID = std::numeric_limits<uint64_t>::max();
```

Comment:
rather than considering 0 as the invalid id, you can define constexpr auto InvalidAgentID = std::numeric_limits<uint64_t>::max();   in types.h use this variable rather than magic number 0 everywhere.